### ISSUE TITLE
Segues in NavigationController viewDidLoad are not captured by Fleet

### ISSUE THREAD

**Issue by jgreubel** [jwfriese/Fleet#8]
This is more of a potential bug.  I would of investigated it a lot more, but I don't have much experience with swizzling.  So, we are probably doing this completely wrong on our project, but we have code in our NavigationController (that is our initialViewController for our main Storyboard).  Specifically it starts off a network call (unrelated, but it does some work) and without waiting for the response from that call, it segues to another viewController.  

Test Code:

```
beforeEach {
    subject = mainStoryboard.instantiateViewControllerWithIdentifier("NavigationController") as! NavigationController
    subject.menuBroker = fakeMenuBroker

    let window = UIWindow()
    window.rootViewController = subject
    window.makeKeyAndVisible()
}

.......

describe("#viewDidLoad") {
    it("begins the fetch for the store menu and store") {
        expect(fakeMenuBroker.updateStoreMenuAndStoreByLocationCallCount).to(equal(1))
    }

    it("saves the store menu and store fetch future") {
        expect(subject.promiseToSetMenuAndStore).to(beIdenticalTo(menuBrokerPromise.future))
    }

    it("segues to the loading view") {
        let expectedLoadingViewController = Fleet.getCurrentScreen()?.topmostPresentedViewController as! LoadingViewController
        expect(expectedLoadingViewController).to(beIdenticalTo(loadingViewController))
    }
}
```

Controller Code:

```
override func viewDidLoad() {
    super.viewDidLoad()

    promiseToSetMenuAndStore = menuBroker.updateStoreMenuAndStoreByLocation()
    performSegueWithIdentifier(Segues.loading, sender: nil)
}
```

The `window.makeKeyAndVisible()` triggers the viewDidLoad and viewWillAppear on the viewController. Here's what happens in test:
1.  Before window.makeKeyAndVisible: Fleet.getCurrentScreen()?.topMostPresentedViewController points to some random UIViewController, no idea what it is....
2.  After window.makeKeyAndVisible: Fleet.getCurrentScreen()?.topMostPresentedViewController points to the NavigationController, even though it performs a segue in the viewDidLoad 

My current workaround is to not use Fleet for this test, and instead global extend the controller and override the performWithSegue to update a global variable that can be checked in test....

**Comment by jgreubel** [comment]
On second note.  This may be unrelated to the viewDidLoad because we have a separate function on our NavigationController that does a segue and Fleet also never gets updated....  So maybe it is more of a NavigationController issue....

```
func returnToMenu(resetScrollPosition: Bool) {
    if resetScrollPosition {
        (viewControllers.first as! MenuViewController).reset()
    }

    popToRootViewControllerAnimated(true)
}
```

Now I am guessing that Fleet maybe doesn't swizzle/catch the .popToRootViewControllerAnimated calls.

**Comment by jwfriese** [comment]
@jgreubel - You're right, Fleet doesn't touch navigation controllers at the moment. It's in the backlog. I'll bump it up since you're running into trouble with it. 

Find the Tracker story here: https://www.pivotaltracker.com/story/show/128851059

I should be able to crack at this soon.

**Comment by jwfriese** [comment]
@jgreubel - Thanks again for filing this issue. The change that I believe fixes your use case can be found in [0.2.1](https://github.com/jwfriese/Fleet/releases/tag/0.2.1). I included a test case that tries to reproduce your exact set-up as closely as possible to ensure that the newest Fleet version covers it 👍 

Please don't hesitate to reach out if it turns out you're still having trouble.
